### ISSUE TITLE
glm::rotation() produce non-identity quaternions for equal vectors

### ISSUE THREAD

**Issue by WinterDog** [g-truc/glm#234]
For equal or almost equal vectors (when `cosTheta` is ~= 1.0f) `glm::rotation()` can produce slightly incorrect quaternions. You can easily test it with this code:
```c++
glm::vec3 testVec = glm::normalize(glm::vec3(5.0f, 5.0f, 5.0f));
glm::quat rotQuat = glm::rotation(testVec, testVec); // Result will be (0.0f, 0.0f, 0.0f, 0.99999994f)
```
It is OK in most cases, but there are specific exceptions. I think it would be better to add check to `glm::rotation()` for such cases, something like that:
```c++
if (cosTheta >= static_cast<T>(1) - epsilon<T>())
    return quat();
```
This would also prevent unnecessary extra calculations like `glm::cross()` (which returns zero vector for that case).

**Comment by Groovounet** [comment]
It looks like I missed that bug... sorry about that. It should be fixed in master branch for GLM 0.9.7.1 release.

Thanks,
Christophe

**Comment by OlivierSohn** [comment]
Hello, I think the fix was not correct, as it produced not an identity quaternion but a zero quaternion. https://github.com/g-truc/glm/pull/691 should fix it I think.
